app/assets/javascripts/components/autocomplete.js

```diff
@@ -23,7 +23,17 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     new window.accessibleAutocomplete.enhanceSelectElement({ // eslint-disable-line no-new, new-cap
       selectElement: $select,
       minLength: 3,
-      showNoOptionsFound: true
+      showNoOptionsFound: true,
+      onConfirm: function (value) {
+        var category = $select.getAttribute('data-track-category')
+        var label = $select.getAttribute('data-track-label')
+        if (!category || !label) {
+          return
+        }
+
+        var action = value
```

Comment:
What is the expected value of `action`?

I was hoping to see a unit test which might have given a clue about what values it'd track in different scenarios. But it looks like there isn't a test for this file at all 🙁 and obviously creating one would out-of-scope for this PR.

Looking at the [Accessible Autocomplete docs](https://alphagov.github.io/accessible-autocomplete/#onconfirm-default---), it seems that the `onConfirm` callback will receive an `Object` parameter. But I was expecting `action` to be a `String` for passing in to `GOVUK.analytics.trackEvent`. So based on that, I suspect this tracking might not work as expected 🤔

Reply:
Yeah, lack of test coverage is really an issue for all of these components.

Action in this case action is a String.
The autocomplete returns [whatever the value of that option was when it was passed into it](https://github.com/kevindew/accessible-autocomplete/blob/master/src/autocomplete.js#L302) when it was initialised.
In this case we're using the enhanceSelectElement function which only [passes the option values to the autocomplete](https://github.com/kevindew/accessible-autocomplete/blob/master/src/wrapper.js#L23) which are the Strings.